graphix/types.py

```diff
@@ -1,10 +1,5 @@
 from __future__ import annotations
 
-import annotated_types
-import typing_extensions
-
-PositiveOrNullInt = typing_extensions.Annotated[int, annotated_types.Ge(0)]  # includes 0
-
```

Comment:
Why do we remove this type?

Reply:
Because this definition would wrongly suggest that `PositiveOrNullInt` alone comes with validation: actually it's not unless used with `BaseModel` (am I correct?).
Actually, it was used in `graphix/sim/density_matrix.py` as such.

Reply:
Types in Python don't inherently come with validation, so there's no surprise there. In the case of `graphix/sim/density_matrix.py`, there was indeed an explicit validation. However, you're also right that `mypy` currently ignores such type annotations, which could make it misleading to use such types without tools to enforce or check them.